### ISSUE TITLE
Assembly R-values for foundation walls and slab insulation

### ISSUE THREAD

**Issue by nmerket** [NREL/hescore-hpxml#215]
**Is your feature request related to a problem? Please describe.**

Above grade walls, knee walls, roofs, and attics all now allow `AssemblyRValue` to be specified instead of a `NominalRValue` for each `Layer` (See #160). It's inconsistent that the same methodology isn't applied for foundation walls and slab insulation. 

**Describe the solution you'd like**

Allow the specification of `AssemblyRValue` instead of `NominalRValue` in the same way we do for other surfaces. Do the weighted R-value calcs as they are done for those. 

**Describe alternatives you've considered**

Leave it as it is. However, people are now asking for this feature.

**Comment by shorowit** [comment]
I agree with this for FoundationWalls, but I don't think it should be allowed for Slabs. Slab insulation is almost always along the perimeter of the slab, and because the ground temperatures vary from center to perimeter, you can't calculate an assembly R-value. If someone is providing an assembly R-value for a slab, it's almost certainly wrong. And in those rare cases where there is insulation that does span the entire floor, there is an [HPXML element to capture that](https://hpxml.nrel.gov/datadictionary/3.0.0/Building/BuildingDetails/Enclosure/Slabs/Slab/UnderSlabInsulationSpansEntireSlab).

**Comment by nmerket** [comment]
Ok. I'm fine to omit this for slabs. It does make more sense for foundation walls.
